app/src/main/java/com/smithnoff/mytaskyapp/ui/home/AgendaHomeFragment.kt

```diff
@@ -119,12 +125,42 @@ class AgendaHomeFragment : Fragment() {
         daysAdapter.setDaysOfMonth(daysOfMonth, selectedDay)
     }
 
-    private fun setFullDateText(){
+    private fun setFullDateText() {
         binding.selectedDate.text =
             if (selectedDate.time == currentDate.time)
                 getString(R.string.txt_today)
             else
                 format.format(selectedDate.time)
         binding.monthSelector.text = months[selectedMonth]
     }
+
+    private fun setUserMenuInfo() {
+        binding.userDropdown.text = sessionManager.getSessionInfo().fullName.abbrevName()
+        binding.userDropdown.setOnClickListener {
+            PopupMenu(requireContext(), it).apply {
+                setOnMenuItemClickListener(this@AgendaHomeFragment)
+                inflate(R.menu.menu_logout)
+                show()
+            }
+        }
+    }
+
+    override fun onMenuItemClick(item: MenuItem?): Boolean {
+        when (item?.itemId) {
+            R.id.logout_user -> {
+                sessionManager.logoutUser()
+                findNavController().navigate(R.id.exit_to_login)
+            }
+        }
+        return true
+    }
+}
+
+
+fun String.abbrevName(): String {
+    return if (this.trim().contains(' ')) {
+        "${this.split(' ')[0].first()}${this.split(' ')[1].first()}"
+    } else {
+        this.take(2)
+    }
```

Comment:
Few thoughts:
1. What if the name is just `Philipp` for example? 
2. What if it contains a middle name like `Philipp Middle Lackner`?

I'd say in (1), the initials should be `PH` and in (2) they should be `PL`. Your code would likely crash in these scenarios or at least behave incorrectly